### PR TITLE
Makefile, scripts, and readme

### PR DESCRIPTION
**Pull request by alexmatson-da**
This PR does a few things - 

1. Tidies up the Makefile, and uses `ddit` for building the package. This will help us automate releases later. 
2. Modifies the circleCI config to call the new `test` targets from the makefile 
3. Adds a script called `verify-versions.sh` which takes the dabl-meta.yaml version as the canonical app version, and checks that all the app versions supplied in other configuration files match up with it. 
4. Starts rewriting the Readme, splits up dev instructions into separate linked docs

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,227 +1,120 @@
...
-### DA Marketplace <> Exberry Adapter
-$(exberry_adapter_dir):
+# Exberry Adapter target
+$(exberry_adapter): $(PKG_DIR)
 	cd exberry_adapter && $(PYTHON) setup.py sdist
 	rm -fr exberry_adapter/marketplace_exchange_adapter.egg-info
+	cp exberry_adapter/dist/$(exberry_adapter_name) $@
 
-$(adapter_pid): |$(state_dir) $(exberry_adapter_dir)
-	cd exberry_adapter && pipenv install && (DAML_LEDGER_URL=localhost:6865 $(PYTHON) bot/exberry_adapter_bot.py > ../$(adapter_log) & echo "$$!" > ../$(adapter_pid))
```
Comment (alexgraham-da) [inline Makefile:149]: I think we should retain this, or add it to the `scripts` folder - maybe also add a section in the local development README about running the adapter/integration? we could always do this in a later PR

File: docs/local_development.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+make stop_matching_engine
+```
+
+Alternatively to running the Daml server in the background, `daml start` can be run in a separate terminal window.
+
+Running the bots manually can be done with:
+
+```
+cd {bot_folder}
+(DAML_LEDGER_URL=localhost:6865 poetry run python bot/{bot_name}_bot.py
```
Comment (alexgraham-da) [inline docs/local_development.md:92]: This is out of date, also see comment in `Makefile` regarding the adapter

File: scripts/verify-versions.sh [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+return_code=0
+
+green='\033[0;92m'
+red='\033[0;91m'
+nc='\033[0m' # No Color
+
+decorate_error () {
+  local msg=$1
+  local file=$2
+  printf "${msg}${red}${file}${nc}\n"
```
Comment (alexgraham-da) [inline scripts/verify-versions.sh:17]: nice 😄

File: docs/damlhub_deployment.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
+[← back](../README.md)
+
+# Deploying to Daml Hub
+
+These instructions will show you how to build and deploy the Marketplace to the Daml Hub cloud service.
+
+## Requirements
+
+To run the marketplace locally or to build to be deployed on Daml Hub, the following requirements are necessary:
```
Comment (Sarahb-da) [inline docs/damlhub_deployment.md:9]: ..`or to build it to be`

File: docs/damlhub_deployment.md [hunk 5]
```diff
@@ -0,0 +1,140 @@
+[← back](../README.md)
```
Comment (alexgraham-da) [inline docs/damlhub_deployment.md:1]: I'm not sure we should hold up the merge rewriting this, but the majority of this is out of date

File: docs/damlhub_deployment.md [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+`da-marketplace-ui-0.0.X.zip`
+`da-marketplace-model-0.0.X.dar`
+
+Click "Launch" for both the UI and the Model:
+
+![4_launch_dar](https://user-images.githubusercontent.com/71082197/98964004-4c802580-24d6-11eb-83bf-221a9d2ce9fa.png)
+
+### Add the parties
+
+Next, add the parties. Click on the "Live Data" tab, and first add the `Public` and `UserAdmin` parties by clicking the "Plus" next to their names:
```
Comment (Sarahb-da) [inline docs/damlhub_deployment.md:57]: this is now the `Identities` tab

File: docs/damlhub_deployment.md [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+
+![5_add_public](https://user-images.githubusercontent.com/71082197/98857333-ed1a0b00-242c-11eb-809a-a7a8a5e983ef.png)
+
+Add the following parties by clicking the "+ Add Party" button: `Alice`, `Bob`, `Exchange`, `Broker`, `UsdtIssuer`, `BtcIssuer` and `Custodian`:
+
+![6_add_parties](https://user-images.githubusercontent.com/71082197/98857334-ed1a0b00-242c-11eb-8f6a-4e385baca6f8.png)
+
+### Bootstrap with example data (optional)
+
+Click the "Ledger Settings" tab, and download the `participants.json` file:
```
Comment (Sarahb-da) [inline docs/damlhub_deployment.md:67]: also in `Identities` tab

File: docs/damlhub_deployment.md [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+Then "Configure New Deployment":
+![t03_configure_new](https://user-images.githubusercontent.com/71082197/101640602-b371fb00-39fe-11eb-9248-2e4d8938d275.png)
+
+### View UI
+
+After all deployments are running, you can click the "View Site" button to visit the Marketplace UI:
+
+![11_view_site](https://user-images.githubusercontent.com/71082197/98857340-edb2a180-242c-11eb-9989-55aafc66199f.png)
+
+The Party ID/Party JWTs to use to login with various parties can be found on the "Users" tab of the console:
```
Comment (Sarahb-da) [inline docs/damlhub_deployment.md:124]: "Identities"

File: docs/local_development.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+
+```
+# start the Daml server
+make start_daml_server
+
+# wait for the navigator to pop up...
+
+# start all bots
+make start_bots
+
```
Comment (Sarahb-da) [inline docs/local_development.md:56]: `make start_autoapprove`?

Comment (Sarahb-da) [reply docs/local_development.md:56]: (and add `make stop_autoapprove`)

**Review by Sarahb-da — APPROVED** [review]
some small stuff

**Review by alexgraham-da — APPROVED** [review]
a few comments, but otherwise this is great!

File: Makefile (hunk 1 above)
Comment (alexmatson-da) [reply Makefile:149]: Good catch, added that back

**Pull request merged**